Assets/Scripts/Boat/BoatMovement.cs

```diff
@@ -0,0 +1,129 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Unity.Mathematics;
+using UnityEngine;
+
+public class BoatMovement : MonoBehaviour
+{
+    [Tooltip("Boat Movement Speed in m/s")]
+    public float maxSpeed;
+    
+    [ReadOnly(true)][Tooltip("The current speed the boat is moving in m/s")]
+    public float currentSpeed;
+
+    [Tooltip("Boat Acceleration in m/s")]
+    public float acceleration;
+    
+    [Tooltip("Boat rotational speed in degrees per second.")]
+    public float rotationSpeed;
+    
+    [Tooltip("Boat rotational limits in degrees. X: Clockwise Limit, Y: AntiClockwise Limit")]
+    public Vector2 rotationLimits;
+
+    public Vector3 currentDirection;
+    [SerializeField] private bool isMoving = false;
+    
+    public KeyCode rotateLeftKeyCode;
+    public KeyCode rotateRightKeyCode;
+    
+    
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        //Only move if the current state is ferrying or returning
+        if (isMoving)
+        {
+            CalculateBoatRotation();
+            CalculateBoatMovement();
+        }
+    }
+
+    public void EnableMovement()
+    {
+        if (!isMoving)
```

Comment:
What side-effect occurs if this check is removed? Any? Might be good to comment it in case someone tries to remove this check